src/subscript/eclcompress/eclcompress.py

```diff
@@ -145,6 +152,20 @@ def eclcompress(files, keeporiginal=False, dryrun=False, eclkw_regexp=None):
     return totalsavings
 
 
+def file_is_binary(filename: Union[str, Path]) -> bool:
+    """Determine if a file is assumed binary
+
+    This uses the same heuristic as the Linux `file` command. Only the first
+    1024 bytes are looked at.
+
+    Args:
+        filename: File to check
+    """
+    # https://stackoverflow.com/questions/898669/how-can-i-detect-if-a-file-is-binary-non-text-in-python  # noqa
+    textchars = bytearray({7, 8, 9, 10, 12, 13, 27} | set(range(0x20, 0x100)) - {0x7F})
+    return bool(open(filename, "rb").read(1024).translate(None, textchars))
```

Comment:
```suggestion
    with open(filename, "rb") as fh:
        return bool(fh.read(1024).translate(None, textchars))
```
...in order to properly close the file handle?